dbms/src/Storages/Page/V3/BlobStore.cpp

```diff
@@ -561,10 +567,74 @@ void BlobStore::read(BlobFileId blob_id, BlobFileOffset offset, char * buffers,
 }
 
 
+struct BlobStoreGCInfo
+{
+    String toString() const
+    {
+        return fmt::format("{}. {}. {}. {}. ",
+                           toTypeString("Read-Only Blob", 0),
+                           toTypeString("No GC Blob", 1),
+                           toTypeString("Full GC Blob", 2),
+                           toTypeString("Truncated Blob", 3));
+    }
+
+    void appendToReadOnlyBlob(const BlobFileId blob_id, double valid_rate)
+    {
+        blob_gc_info[0].emplace_back(std::make_pair(blob_id, valid_rate));
+    }
+
+    void appendToNoNeedGCBlob(const BlobFileId blob_id, double valid_rate)
+    {
+        blob_gc_info[1].emplace_back(std::make_pair(blob_id, valid_rate));
+    }
+
+    void appendToNeedGCBlob(const BlobFileId blob_id, double valid_rate)
+    {
+        blob_gc_info[2].emplace_back(std::make_pair(blob_id, valid_rate));
+    }
+
+    void appendToTruncatedBlob(const BlobFileId blob_id, double valid_rate)
+    {
+        blob_gc_info[3].emplace_back(std::make_pair(blob_id, valid_rate));
+    }
+
+private:
+    // 1. read only blob
+    // 2. no need gc blob
+    // 3. full gc blob
+    // 4. need truncate blob
+    std::vector<std::pair<BlobFileId, double>> blob_gc_info[4];
+
+    String toTypeString(const String prefix, const size_t index) const
+    {
+        FmtBuffer fmt_buf;
+
+        fmt_buf.fmtAppend("{}: [", prefix);
+        if (blob_gc_info[index].empty())
+        {
+            fmt_buf.append("null]");
+        }
+        else
+        {
+            fmt_buf.joinStr(
+                blob_gc_info[index].begin(),
+                blob_gc_info[index].end(),
+                [](const auto arg, FmtBuffer & fb) {
+                    fb.fmtAppend("{}/{:.2f}", arg.first, arg.second);
+                },
+                ", ");
+            fmt_buf.append("]");
+        }
```

Comment:
suggest make it a simple one func call for empty
```suggestion
        if (blob_gc_info[index].empty())
        {
            fmt_buf.fmtAppend("{}: [null]", prefix);
        }
        else
        {
            fmt_buf.fmtAppend("{}: [", prefix);
            fmt_buf.joinStr(
                blob_gc_info[index].begin(),
                blob_gc_info[index].end(),
                [](const auto arg, FmtBuffer & fb) {
                    fb.fmtAppend("{}/{:.2f}", arg.first, arg.second);
                },
                ", ");
            fmt_buf.append("]");
        }
```